core/arch/arm/include/sm/optee_smc.h

```diff
@@ -290,6 +290,12 @@
 #define OPTEE_SMC_SEC_CAP_HAVE_RESERVED_SHM	(1 << 0)
 /* Secure world can communicate via previously unregistered shared memory */
 #define OPTEE_SMC_SEC_CAP_UNREGISTERED_SHM	(1 << 1)
+/*
```

Comment:
What do you mean with respect to "negotiated"?  UNREGISTERED_SHM is unused as of now in both OP-TEE and Linux.

This is how the capabilities are defined today:

/* Secure world has reserved shared memory for normal world to use */
#define OPTEE_SMC_SEC_CAP_HAVE_RESERVED_SHM     (1 << 0)
/* Secure world can communicate via previously unregistered shared memory */
#define OPTEE_SMC_SEC_CAP_UNREGISTERED_SHM      (1 << 1)

It is HAVE_RESERVED_SHM that is referring to the physically contiguous static CFG_SHMEM_START/SIZE.

The question is what capabilities we need to refer to the new ability for OP-TEE to map arbitrary, non-contiguous physical addresses.

Do we need:
A).  One capability that refers to the capability of OP-TEE to handle non-contiguous shared memory and the capability to handle the register/unregister core APIs?
or
B) . Two capabilities-- 1) one for the capability of OP-TEE to handle non-contiguous shared memory, and 2) one for the capability to handle the register/unregister core APIs.

Jens proposal on the mailing list was for option #A.